src/Maker/MakeEntity.php

```diff
@@ -317,10 +318,17 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
     public function configureDependencies(DependencyBuilder $dependencies, InputInterface $input = null): void
     {
         if (null !== $input && $input->getOption('api-resource')) {
-            $dependencies->addClassDependency(
-                ApiResource::class,
-                'api'
-            );
+            if (class_exists(LegacyApiResource::class)) {
```

Comment:
I would test for `ApiResource` first because in 2.7 both classes exist, and we prefer using the non-deprecated attribute.

Reply:
Hi Kévin, I fixed that.